React/Base/RCTViewRegistryPair.h

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "RCTComponentRetrievable.h"
+#import "RCTPair.h"
+
+@interface RCTViewRegistryPair<RCTComponentRetrievable> : RCTPair<UIView *, UIViewController *>
```

Comment:
You're using Pair to mean it's either a view or a viewcontroller. The correct meaning of a pair though is a tuple of a view _and_ a view controller.